apps/web/src/dialogs/command-palette-dialog.tsx

```diff
@@ -0,0 +1,223 @@
+    action: () => navigate("/notebooks")
+  },
+  {
+    title: "Go to tags",
+    icon: ArrowTopRight,
+    action: () => navigate("/tags")
+  },
+  {
+    title: "Go to favorites",
+    icon: ArrowTopRight,
+    action: () => navigate("/favorites")
+  },
+  {
+    title: "Go to reminders",
+    icon: ArrowTopRight,
+    action: () => navigate("/reminders")
+  },
+  {
+    title: "Go to monographs",
+    icon: ArrowTopRight,
+    action: () => navigate("/monographs")
+  },
+  {
+    title: "Go to trash",
+    icon: ArrowTopRight,
+    action: () => navigate("/trash")
+  },
+  {
+    title: "Go to settings",
+    icon: ArrowTopRight,
+    action: () => hashNavigate("/settings", { replace: true })
+  },
+  {
+    title: "Go to help",
+    icon: ArrowTopRight,
+    action: () => (window.location.href = "https://help.notesnook.com")
+  },
+  {
+    title: "New note",
+    icon: Plus,
+    action: () => useEditorStore.getState().newSession()
+  },
+  {
+    title: "New notebook",
+    icon: Plus,
+    action: () => hashNavigate("/notebooks/create", { replace: true })
+  },
+  {
+    title: "New tag",
+    icon: Plus,
+    action: () => hashNavigate("/tags/create", { replace: true })
+  },
+  {
+    title: "New reminder",
+    icon: Plus,
+    action: () => hashNavigate(`/reminders/create`, { replace: true })
+  },
+  {
+    title: "New color",
+    icon: Plus,
+    action: () => CreateColorDialog.show(true)
+  },
+  {
+    title: "Toggle theme",
+    icon: Radar,
+    action: () => useThemeStore.getState().toggleColorScheme()
+  }
+];
+
+export const CommandPaletteDialog = DialogManager.register(
+  function CommandPaletteDialog(props: BaseDialogProps<boolean>) {
+    const [filteredCommands, setFilteredCommands] = useState(commands);
+    const [selected, setSelected] = useState<number>(0);
+    const selectedRef = useRef<HTMLButtonElement>(null);
+
+    useEffect(() => {
+      selectedRef.current?.scrollIntoView({
+        block: "nearest"
+      });
+    }, [selected]);
+
+    return (
+      <Dialog
+        isOpen={true}
+        width={650}
+        onClose={() => props.onClose(false)}
+        noScroll
+        sx={{
+          fontFamily: "body"
+        }}
+      >
+        <Flex
+          variant="columnFill"
+          sx={{ mx: 3, overflow: "hidden", height: 450 }}
+          onKeyDown={(e) => {
+            if (e.key == "Enter") {
+              e.preventDefault();
+              if (filteredCommands[selected]) {
+                filteredCommands[selected].action();
+                setFilteredCommands(commands);
+                setSelected(0);
+                props.onClose(true);
+              } else {
+                setSelected(0);
+              }
+            }
+            if (e.key === "ArrowDown") {
+              e.preventDefault();
+              setSelected(
+                (selected) => (selected + 1) % filteredCommands.length
+              );
+            }
+            if (e.key === "ArrowUp") {
```

Comment:
I suggest you keep these keyboard keys as an enum in a separate file to avoid using "[magic strings](https://stoic.software/articles/magic-strings-the-anti-pattern/)".

Reply:
Sounds good, maybe we can take this up in a separate PR